### PR TITLE
Add blob example

### PR DESCRIPTION
**Pull request by dfaust**
It took me a while to figure out that calling `into()` on a `Vec<u8>` in order to convert it into `Bytes` doesn't work as I expected.
`impl<T: Into<Bytes> + Clone + Debug> From<Vec<T>> for Bytes` exists but I'm not sure what it does exactly.

### CODE REVIEW

**Comment by AlexPikalov** [comment]
@dfaust `impl<T: Into<Bytes> + Clone + Debug> From<Vec<T>> for Bytes` converts a list of values into `Bytes` which is one of steps to convert into `Value` and then pass into query builder. For blob type which is already an array of bytes we can create a value directly `Value::new_normal(blob)`. To be consistency with other type we may want to implement `impl Into<Bytes> for Vec<u8>` simply as

```rust
impl Into<Bytes> for Vec<u8> {
    fn into(self) -> Bytes {
        Bytes(self)
    }
}
```
This would be enough to be able to call `vec![1,2,3,4].into()`

File: examples/all.rs [hunk 1]
```diff
@@ -589,6 +606,41 @@ fn select_table_float(session: &mut Session<NoneAuthenticator, TransportTcp>) ->
...
+    let q = QueryBuilder::new(CREATE_TABLE_BLOB).finalize();
+    match session.query(q, false, false) {
+        Err(ref err) => panic!("create_table blob {:?}", err),
+        Ok(_) => true,
+    }
+}
+
+fn insert_table_blob(session: &mut Session<NoneAuthenticator, TransportTcp>) -> bool {
+    let blob: Vec<u8> = vec![0, 1, 2, 4, 8, 16, 32, 64, 128, 255];
+    let values: Vec<Value> = vec![(1 as i32).into(), Bytes::new(blob).into()];
```
Comment (AlexPikalov) [inline examples/all.rs]: yeah. this approach could also be an option

**Comment by AlexPikalov** [comment]
as for 
```
impl Into<Bytes> for Vec<u8> {
    fn into(self) -> Bytes {
        Bytes(self)
    }
}
```
I've just checked it raises a compilation error as a conflicting implementation

**Comment by dfaust** [comment]
Yep, this needs specialization (https://github.com/rust-lang/rfcs/pull/1210/) which is still behind a feature flag on nightly rust.

**Comment by dfaust** [comment]
And what I don't understand is, why `impl<T: Into<Bytes> + Clone + Debug> From<Vec<T>> for Bytes` doesn't work right now. If it was doing what I expected, it should convert the `Vec<u8>` into `Bytes` correctly but with a lot of unnecessary overhead. But it doesn't work, the data gets altered.

**Comment by AlexPikalov** [comment]
In case of blob or `Vec<u8>` in other words what we actually need is to pass an array of bytes into `Value`. Value is a 4-bytes length and array of bytes. What `impl<T: Into<Bytes> + Clone + Debug> From<Vec<T>> for Bytes` does is maps a list of separate values into `Value` one by one. Then concatenates bytes into a single byte array. Then basing of this array method builds non-empty value.

In case of blob (`Vec<u8>`) what we want to have is `vec![1, 2] -> vec![0, 0, 0, 2, 1, 2]`. And what `impl<T: Into<Bytes> + Clone + Debug> From<Vec<T>> for Bytes` does is `vec![1, 2] -> vec![0, 0, 0, 1, 1, 0, 0, 0, 1, 2] -> vec![0, 0, 0, 10, 0, 0, 0, 1, 1, 0, 0, 0, 1, 2]`

**Comment by AlexPikalov** [comment]
So there is an issue here. That for blob we should consider `Vec<u8>` as a single value, while in other cases ie. `Vec<String>` should be considered as a set of values (Cassandra's list or set type)

**Comment by AlexPikalov** [comment]
Yes, looking into mapping table I was thinking about the same. Probably `Into<u8>` is useless because of reasons you mentioned. So removing its implementation would allow to add without conflicts

```
impl Into<Bytes> for Vec<u8> {
    fn into(self) -> Bytes {
        Bytes(self)
    }
}
```

**Pull request merged**